lib/src/hex_util.rs

```diff
@@ -51,6 +51,13 @@ pub fn to_reverse_hex(forward_hex: &str) -> Option<String> {
         .collect()
 }
 
+pub fn decode_hex_string(hex: &str) -> Option<Vec<u8>> {
```

Comment:
Can you reorder (or squash) the first two patches? It's easier to follow if we had a drop-in replacement for `hex::decode()`.